packages/web-components/src/components/ic-accordion-group/ic-accordion-group.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  Component,
+  Element,
+  h,
+  Host,
+  Listen,
+  Prop,
+  State,
+} from "@stencil/core";
+import { IcThemeForeground, IcSizes } from "../../utils/types";
+
+let accordionGroupIds = 0;
+@Component({
+  tag: "ic-accordion-group",
+  styleUrl: "ic-accordion-group.css",
+  shadow: true,
+})
+export class AccordionGroup {
+  private accordionGroupId = `ic-accordion-group-${accordionGroupIds++}`;
+
+  @Element() el: HTMLIcAccordionGroupElement;
+
+  @State() accordions: HTMLIcAccordionElement[];
+
+  @State() areAllAccordionsOpen: boolean;
+  /**
+   * The appearance of the tab context, e.g dark, or light.
+   */
+  @Prop() appearance: IcThemeForeground = "default";
+
+  /**
+   * If `true`, the side navigation will load in an expanded state.
+   */
+  @Prop({ mutable: true }) expanded: boolean = false;
+
+  /**
+   * The header for the accordion group.
+   */
+  @Prop() groupTitle: string = "";
+
+  /**
+   * If `true`, only one accordion will open at a time.
+   */
+  @Prop() singleExpansion: boolean = false;
+
+  /**
+   * The size of the accordion.
+   */
+  @Prop() size?: IcSizes = "default";
+
+  componentDidLoad(): void {
+    const accordionDirectChildren = (this.el as HTMLElement).children;
+    this.accordions = Array.from(accordionDirectChildren).filter(
+      (child) => child.tagName === "IC-ACCORDION"
+    ) as HTMLIcAccordionElement[];
+    this.linkAccordions();
+    this.accordions.forEach((acc) => {
+      acc.appearance = this.appearance;
+    });
+    this.accordions.forEach((acc) => {
+      acc.size = this.size;
+    });
+    if (this.expanded) {
+      this.accordions.forEach((acc) => {
+        acc.expanded = true;
+      });
+      this.setExpandedToAreAllAccordionsOpen();
+    } else {
+      this.setExpandedToAreAllAccordionsOpen();
+      this.expanded = this.areAllAccordionsOpen;
+    }
+  }
+
+  @Listen("accordionClicked")
+  handleAccordionClicked(event: CustomEvent) {
+    if (this.singleExpansion) {
+      this.accordions.forEach((acc) => {
+        acc.expanded = false;
+      });
+      const singleAccordion = this.accordions.find(
+        (acc) => acc.id === event.detail.id
+      );
+      singleAccordion.expanded = true;
```

Comment:
I may have missed this part of the AC, but when singleExpansion is enabled, I'm seeing behaviour where I can't close the one i've opened once it has been opened. Is this correct? My understanding was that only one could be open at once, but all could be closed at the same time. Also noticing some flickering when clicking the other accordion when it closes them all.
[Screencast from 18-07-23 10:47:54.webm](https://github.com/mi6/ic-ui-kit/assets/117368893/dc91fd63-6237-4a98-947c-ef6b727e7978)

Reply:
I think if this is the correct behaviour maybe hiding the closing animation that appears before it immediately opens again, as it may throw people off and have them thinking it didn't register their clicks.